Older-version snapshots and repositories are guaranteed to work with … (#7275)
opensearch-project/documentation-website: _tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore.md

1. A write request to the now-deleted alias creates a new index with the same name as the alias.
1. The alias from the snapshot fails to restore due to a naming conflict with the new index.

> Snapshots are only forward compatible by one major version. Snapshots taken by earlier OpenSearch versions can continue to be restored by the version of OpenSearch that originally took the snapshot, even after a version upgrade. For example, a snapshot taken by OpenSearch 2.11 or earlier can continue to be restored with 2.11 clusters even after upgrading to 2.12.

Reviewer: ```suggestion
Snapshots are only forward compatible by one major version. Snapshots taken by earlier OpenSearch versions can continue to be restored by the version of OpenSearch that originally took the snapshot, even after a version upgrade. For example, a snapshot taken by OpenSearch 2.11 or earlier can continue to be restored by a 2.11 cluster even after upgrading to 2.12.
```

1. A write request to the now-deleted alias creates a new index with the same name as the alias.
1. The alias from the snapshot fails to restore due to a naming conflict with the new index.

> Snapshots are only forward-compatible by one major version. Snapshots taken by older OpenSearch versions can continue to be restored by the version of OpenSearch that originally took the snapshot even after a version upgrade. For example - a snapshot taken from OpenSearch 2.11.0 or lower can be used for restores with 2.11.0 clusters even after upgrading to 2.12.0.

Reviewer 2: ```suggestion
Snapshots are only forward-compatible by one major version. Snapshots taken by older OpenSearch versions can continue to be restored by the version of OpenSearch that originally took the snapshot even after a version upgrade. For example - a snapshot taken from OpenSearch 2.11.0 or lower can be used snapshot restores with 2.11.0 clusters even after upgrading to 2.12.0.
```

> Snapshots are only forward-compatible by one major version. Snapshots taken by older OpenSearch versions can continue to be restored by the version of OpenSearch that originally took the snapshot even after a version upgrade. For example - a snapshot taken from OpenSearch 2.11.0 or lower can be used snapshot restores with 2.11.0 clusters even after upgrading to 2.12.0.

Reviewer: ```suggestion
Snapshots are only forward compatible by one major version. Snapshots taken by earlier OpenSearch versions can continue to be restored by the version of OpenSearch that originally took the snapshot, even after a version upgrade. For example, a snapshot taken by OpenSearch 2.11 or earlier can continue to be restored even after upgrading to 2.12.
```

> Snapshots are only forward-compatible by one major version. Snapshots taken by older OpenSearch versions can continue to be restored by the version of OpenSearch that originally took the snapshot even after a version upgrade. For example - a snapshot taken from OpenSearch 2.11.0 or lower can be used snapshot restores with 2.11.0 clusters even after upgrading to 2.12.0.

Reviewer: Confirm that my changes to the last sentence retain technical accuracy.

> If you have an old snapshot taken from an older major OpenSearch version, you can restore it into an intermediate cluster on one major version newer than the snapshot's version, reindex all indexes, take a new snapshot, and repeat until you arrive at your desired major version, but you might find it easier to manually index your data in the new cluster.

Reviewer: ```suggestion
If you have an old snapshot taken from an earlier major OpenSearch version, you can restore it to an intermediate cluster one major version newer than the snapshot's version, reindex all indexes, take a new snapshot, and repeat until you arrive at your desired major version, but you may find it easier to manually index your data in the new cluster.
```

> If you have an old snapshot taken from previous major OpenSearch versions, you can restore it into an intermediate cluster on one major version newer than the snapshot's version, reindex all indexes, take a new snapshot, and repeat until you arrive at your desired major version, but you might find it easier to just manually index your data in the new cluster.

Reviewer 2: ```suggestion
If you have an old snapshot taken from an older major OpenSearch version, you can restore it into an intermediate cluster on one major version newer than the snapshot's version, reindex all indexes, take a new snapshot, and repeat until you arrive at your desired major version, but you might find it easier to just manually index your data in the new cluster.
```

Snapshots are only forward-compatible by one major version. If you have an old snapshot, you can sometimes restore it into an intermediate cluster, reindex all indexes, take a new snapshot, and repeat until you arrive at your desired version, but you might find it easier to just manually index your data in the new cluster.

> Older-version snapshots and the repositories that contain them continue to be compatible with older versions of OpenSearch. If you have a repository containing snapshots taken by an old version, you then upgrade to another version and snapshots are taken after that as well. The old snapshots can be restored to a new cluster running the same version.

Reviewer 2: Thanks for the submission @tan3-netapp, are there limitations to using older versioned snapshots in newer versions of OpenSearch? Is it restricted to major versions, for example?

Author: I made an update here after encountering this [issue](https://github.com/opensearch-project/OpenSearch/issues/13676). What I updated is just a guarantee that older versioned snapshots can always be restored to a new cluster running the same version even if the snapshot repository has been modified by the new version. (as I put it in the **Question 2** in the [GitHub issue](https://github.com/opensearch-project/OpenSearch/issues/13676) mentioned above).

Anyway, I've changed my wording. Please have another look. Thanks!
